src/element/writer.rs

```diff
@@ -12,10 +12,17 @@ pub use TextKind::*;
 
 /// Serializes [`Element`] instances into some kind of output sink.
 pub trait ElementWriter {
-    /// Serializes a single [`Element`] as a top-level value.
+    /// Serializes a single [`Element`] at the current depth of the writer.
     fn write_element(&mut self, element: &Element) -> IonResult<()>;
 
-    /// Serializes a collection of [`Element`] as a series of top-level values.
+    /// Serializes a single [`Value`] at the current depth of the writer.
+    // TODO: consider extracting this to a ValueWriter trait.
+    fn write_value(&mut self, value: &Value) -> IonResult<()>;
```

Comment:
This makes me wonder if we need/want a `write_values(iter)` sibling method, but I'm happy to punt on that.